### ISSUE TITLE
Nan mIoU on "voc 19-1 step1" task under "Sequential" setting

### ISSUE THREAD

**Issue by SSSKYue** [LTTM/SDR#6]
Hello, thanks for your innovative method and such a good work!
Sorry to bother you. I got a similar result as yours under the disjoint setting, but when I try to reproduce the 'Sequential' tasks, I got a weird nan mIoU and class loss 0.0 on the validation of  "voc 19-1 step1".  
I noticed the **.npy** files have the same image indexes between "voc/19-1/" and"voc/19-1-oldclICCVW2019"，
I train the step1 model using just the same step0 model trained on 'disjoint 19-1 step0' settings, and my command is:

python -u -m torch.distributed.launch 1> 'outputs/19-1/output_19-1_step1_LWF_seq.txt'  2>&1 --nproc_per_node=1 run.py --batch_size 8 --logdir logs/19-1/ --dataset voc --name LWF_seq --task 19-1 --step 1 --lr 0.0001 --epochs 30 --debug --sample_num 10 --where_to_sim GPU_windows --method LWF --step_ckpt 'logs/19-1/19-1-voc_FT/19-1-voc_FT_0.pth' --no_mask

the loss during training seems working normally but the validation outputs is:
INFO:rank0: Validation, Class Loss=0.0, Reg Loss=0.011368847452104092 (without scaling)
INFO:rank0: Done validation on Val set
INFO:rank0: End of Validation 15/30, Validation Loss=0.011368847452104092, Class Loss=0.0, Reg Loss=0.011368847452104092
INFO:rank0: 
Total samples: 74.000000
Overall Acc: nan
Mean Acc: nan
FreqW Acc: 0.000000
Mean IoU: nan
Class IoU:
	class 0: X
	class 1: X
	class 2: X
	class 3: X
	class 4: X
	class 5: X
	class 6: X
	class 7: X
	class 8: X
	class 9: X
	class 10: X
	class 11: X
	class 12: X
	class 13: X
	class 14: X
	class 15: X
	class 16: X
	class 17: X
	class 18: X
	class 19: X
	class 20: X
Class Acc:
	class 0: X
	class 1: X
	class 2: X
	class 3: X
	class 4: X
	class 5: X
	class 6: X
	class 7: X
	class 8: X
	class 9: X
	class 10: X
	class 11: X
	class 12: X
	class 13: X
	class 14: X
	class 15: X
	class 16: X
	class 17: X
	class 18: X
	class 19: X
	class 20: X

I didn't modified your code. Could you give me some comments？Thank you very much!

**Comment by umbertomichieli** [comment]
Hi, thank you for pointing this out. There was indeed a mistake in the way data was processed and the labels were all 255. Please download the last version.
The only file that changes is: _dataset/voc.py_

Kind regards

**Comment by SSSKYue** [comment]
I see, thanks a lot!

**Comment by ljx97** [comment]
Hello,
Sorry to bother you. I can't get a similar result as the article under the disjoint setting of voc 15-5s tasks. Would you please tell me what command you run? My commands are as follows:

python run.py \
--batch_size 8 \
--logdir logs/15-5s/ \
--dataset voc \
--name FT \
--task 15-5s \
--step 0 \
--lr 0.001 \
--epochs 30 \
--debug \
--sample_num 10 \
--unce \
--loss_de_prototypes 1 \
--where_to_sim GPU_windows


python run.py \
--batch_size 8 \
--logdir logs/15-5s/ \
--dataset voc \
--name SDR1\
--task 15-5s \
--step 1 \
--lr 0.0001 \
--epochs 30 \
--debug \
--sample_num 10 \
--where_to_sim GPU_windows \
--method SDR \
--step_ckpt 'logs/15-5s/15-5s-voc_FT/15-5s-voc_FT_0.pth'

python run.py \
--batch_size 8 \
--logdir logs/15-5s/ \
--dataset voc \
--name SDR2
--task 15-5s \
--step 2 \
--lr 0.0001 \
--epochs 30 \
--debug \
--sample_num 10 \
--where_to_sim GPU_windows \
--method SDR \
--step_ckpt 'logs/15-5s/15-5s-voc_SDR1/15-5s-voc_SDR1_1.pth'

python run.py \
--batch_size 8 \
--logdir logs/15-5s/ \
--dataset voc \
--name SDR3
--task 15-5s \
--step 3 \
--lr 0.0001 \
--epochs 30 \
--debug \
--sample_num 10 \
--where_to_sim GPU_windows \
--method SDR \
--step_ckpt 'logs/15-5s/15-5s-voc_SDR2/15-5s-voc_SDR2_2.pth'

python run.py \
--batch_size 8 \
--logdir logs/15-5s/ \
--dataset voc \
--name SDR4
--task 15-5s \
--step 4 \
--lr 0.0001 \
--epochs 30 \
--debug \
--sample_num 10 \
--where_to_sim GPU_windows \
--method SDR \
--step_ckpt 'logs/15-5s/15-5s-voc_SDR3/15-5s-voc_SDR3_3.pth'

python run.py \
--batch_size 8 \
--logdir logs/15-5s/ \
--dataset voc \
--name SDR5
--task 15-5s \
--step 5 \
--lr 0.0001 \
--epochs 30 \
--debug \
--sample_num 10 \
--where_to_sim GPU_windows \
--method SDR \
--step_ckpt 'logs/15-5s/15-5s-voc_SDR4/15-5s-voc_SDR4_4.pth'

**Comment by umbertomichieli** [comment]
Hi, the commands look fine to me. But at this time I don't have access to any GPU to test them. I will get back to you once I can test the commands on the proper open issue.

**Comment by ljx97** [comment]
Hello, the results I reproduced of the voc 15-5s task are not fine.  Would you like to share all logs of voc-15-5s task. Thanks a lot.

**Comment by umbertomichieli** [comment]
Hi, let's continue the discussion on the proper issue #8

**Comment by SSSKYue** [comment]
> 

Sorry, I didn't try the '15-5s' task yet, I'll share the results with you when I test it.

**Comment by ljx97** [comment]
Okey. Thanks a lot.
On 12/07/2021 17:18, SSSKYue wrote:

Sorry, I didn't try the '15-5s' task yet, I'll share the results with you when I test it.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
